homeassistant/components/bmw_connected_drive/notify.py

```diff
@@ -0,0 +1,69 @@
+"""Support for BMW notifications."""
+import logging
+
+from homeassistant.components.notify import (
+    ATTR_DATA,
+    ATTR_TARGET,
+    ATTR_TITLE,
+    ATTR_TITLE_DEFAULT,
+    BaseNotificationService,
+)
+from homeassistant.const import ATTR_LATITUDE, ATTR_LOCATION, ATTR_LONGITUDE, ATTR_NAME
+
+from . import DOMAIN as BMW_DOMAIN
+
+ATTR_LAT = "lat"
+ATTR_LOCATION_ATTRIBUTES = ["street", "city", "postal_code", "country"]
+ATTR_LON = "lon"
+ATTR_SUBJECT = "subject"
+ATTR_TEXT = "text"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_service(hass, config, discovery_info=None):
+    """Get the BMW notification service."""
+    accounts = hass.data[BMW_DOMAIN]
+    _LOGGER.debug("Found BMW accounts: %s", ", ".join([a.name for a in accounts]))
+    return BMWNotificationService(accounts)
+
+
+class BMWNotificationService(BaseNotificationService):
+    """Send Notifications to BMW."""
+
+    def __init__(self, accounts):
+        """Set up the notification service."""
+        self.targets = {}
+        for account in accounts:
```

Comment:
Would you please move the side effects out of init.

Maybe a `def setup` that you can call to setup the class before returning it in `get_service`

Reply:
You mean like this?

```py
def get_service(hass, config, discovery_info=None):
    """Get the BMW notification service."""
    accounts = hass.data[BMW_DOMAIN]
    _LOGGER.debug("Found BMW accounts: %s", ", ".join([a.name for a in accounts]))
    svc = BMWNotificationService()
    svc.setup(accounts)
    return svc


class BMWNotificationService(BaseNotificationService):
    """Send Notifications to BMW."""

    def __init__(self):
        """Set up the notification service."""
        self.targets = {}

    def setup(self, accounts):
        """Get the BMW vehicle(s) for the account(s)."""
        for account in accounts:
            self.targets.update({v.name: v for v in account.account.vehicles})
```

Reply:
Updates made.